scripts/exregional_run_prepstart.sh

```diff
@@ -98,27 +98,58 @@ YYYYMMDD=${YYYYMMDDHH:0:8}
 # prepare initial conditions for 
 #     cold start if BKTYPE=1 
 #     warm start if BKTYPE=0
+#     spinupcyc + warm start if BKTYPE=0
 #       the previous 6 cycles are searched to find the restart files
 #       valid at this time from the closet previous cycle.
 #
 #-----------------------------------------------------------------------
 
+BKTYPE=0
+if [ ${cycle_type} -eq 1 ]; then
+   echo "spin up cycle"
+  for cyc_start in ${CYCL_HRS_SPINSTART[@]}; do
+    if [ ${HH} -eq ${cyc_start} ]; then
+      BKTYPE=1
+    fi
+  done
+else
+  echo " product cycle"
+  for cyc_start in ${CYCL_HRS_PRODSTART[@]}; do
+    if [ ${HH} -eq ${cyc_start} ]; then
+      if [ ${DO_SPINUP} == "true" ]; then
+        BKTYPE=2   # using 1-h from spinup cycle
+      else
+        BKTYPE=1
+      fi
+    fi
+  done
+fi
+
 cd_vrfy ${modelinputdir}
 
-if [ ${BKTYPE} -eq 1 ]; then  # cold start, use prepare cold strat initial files from ics
-  bkpath=${cycle_dir}/ics
-  if [ -r "${bkpath}/gfs_data.tile7.halo0.nc" ]; then
-    cp_vrfy ${bkpath}/gfs_bndy.tile7.000.nc gfs_bndy.tile7.000.nc        
-    cp_vrfy ${bkpath}/gfs_ctrl.nc gfs_ctrl.nc        
-    cp_vrfy ${bkpath}/gfs_data.tile7.halo0.nc gfs_data.tile7.halo0.nc        
-    cp_vrfy ${bkpath}/sfc_data.tile7.halo0.nc sfc_data.tile7.halo0.nc        
-    print_info_msg "$VERBOSE" "cold start from $bkpath"
+if [ ${BKTYPE} -eq 1 ] ; then  # cold start, use prepare cold strat initial files from ics
+    bkpath=${cycle_dir}/ics
+    if [ -r "${bkpath}/gfs_data.tile7.halo0.nc" ]; then
+      cp_vrfy ${bkpath}/gfs_bndy.tile7.000.nc gfs_bndy.tile7.000.nc        
+      cp_vrfy ${bkpath}/gfs_ctrl.nc gfs_ctrl.nc        
+      cp_vrfy ${bkpath}/gfs_data.tile7.halo0.nc gfs_data.tile7.halo0.nc        
+      cp_vrfy ${bkpath}/sfc_data.tile7.halo0.nc sfc_data.tile7.halo0.nc        
+      print_info_msg "$VERBOSE" "cold start from $bkpath"
+    else
+      print_err_msg_exit "Error: cannot find cold start initial condition from : ${bkpath}"
+    fi
+else
+
+  if [ ${cycle_type} -eq 1 ]; then
+     fcst_dir_name=fcst_fv3lam_spinup
   else
-    print_err_msg_exit "Error: cannot find cold start initial condition from : ${bkpath}"
+     fcst_dir_name=fcst_fv3lam
+  fi
+  if [ ${BKTYPE} -eq 2 ]; then
+     fcst_dir_name=fcst_fv3lam_spinup
   fi
```

Comment:
The original code is a little bit longer, but it may be easy to understand the directory structure. 

if condition for cycle_type
143   if [ ${cycle_type} -eq 1 ]; then
144      fcst_dir_name=fcst_fv3lam_spinup
145   else
146      fcst_dir_name=fcst_fv3lam
147   fi
# warm start from spinup cycle
149   if [ ${BKTYPE} -eq 2 ]; then
150      fcst_dir_name=fcst_fv3lam_spinup
151   fi

Reply:
In general, code doesn't really get clearer with more logic.

Reply:
Agree and changed.